### ISSUE TITLE
French ReNaSS earthquake network added

### ISSUE THREAD

**Issue by ddperso** [poblabs/weewx-belchertown#561]
Hi

In France we have a dedicated network for earthquake named ReNaSS (Réseau Nationale de Surveillance Sismique)
Like GeoNet, I have added this network to the skin
To do that I have modified belchertown.py
_______________________________________________________________________________________________
            elif self.generator.skin_dict["Extras"]["earthquake_server"] == "ReNaSS":
                earthquake_url = (
                    "https://renass.unistra.fr/fdsnws/event/1/query?latitude=%s&longitude=%s&maxradius=5&orderby=time&format=json&limit=1&mindepth=-1"
                    % (latitude, longitude)
                )
_ _ _ _ _ _  _ _ _ _ _ _ _ _ _ __ _

                elif self.generator.skin_dict["Extras"]["earthquake_server"] == "ReNaSS":
                    eqtime = eqdata["features"][0]["properties"]["time"]
                    # convert time to UNIX format
                    eqtime = datetime.datetime.strptime(eqtime, "%Y-%m-%dT%H:%M:%S.%fZ")
                    eqtime = int(
                        (eqtime - datetime.datetime(1970, 1, 1)).total_seconds()
                    )
                    equrl = eqdata["features"][0]["properties"]["url"]
                    eqplace = eqdata["features"][0]["properties"]["description"]
                    eqmag = locale.format(
                        "%g", float(eqdata["features"][0]["properties"]["mag"])
                    )
_______________________________________________________________________________________________

and add a new value in weewx.conf (like USGS or GeoNet)

_______________________________________________________________________________________________
            earthquake_stale = 3600
            earthquake_server = ReNaSS
_______________________________________________________________________________________________

Sorry, I don't know how to do a pull request.

![seisme](https://user-images.githubusercontent.com/65628453/106740666-fde8b000-661a-11eb-8703-c519c3e5aa10.jpg)

**Comment by norecords** [comment]
Very nice!
Merci ddperso, i'll have look :)

**Comment by norecords** [comment]
@ddperso 
I encourage you to make a pull request by yourself, a good tutorial here
https://opensource.com/article/19/7/create-pull-request-github
In main lines you have to
Fork this repo, then clone it locally, add new branch, make changes, push your forked repo and then create pull request

**Comment by steflp** [comment]
Hello  @ddperso 
Can you tell me ( us) which number line you modify  ? 
Can you put the file ? 
Best  regards

(in french)  :
Pouvez-vous nous donner a partir de quelle ligne vous avez apporté vos modifications  ?
(Peut etre mettre le fichier en pj  ?  ) 
Merci

Stéphane

**Comment by norecords** [comment]
I made a pull request #568

**Comment by norecords** [comment]
You may leave `earthquake_stale` to `10740` (seconds = 179 minutes) if you don't really need to check every hours

**Comment by norecords** [comment]
Just found how to calculate the max radius for ReNaSS api, it seem that we have to divide Km by 111.25
source: https://www.orfeus-eu.org/data/eida/webservices/examples/workflow/webservice_example_urlquery.py
@ line 90

**Comment by norecords** [comment]
@steflp now you can see how to change your file
https://github.com/poblabs/weewx-belchertown/pull/568/files

**Comment by steflp** [comment]
Hello
I ve seen thanks a  lot !
why you change this too ? line  1683 ? 
`if earthquake_maxradiuskm <= "1000":
                earthquake_ReNaSSradius = 5
            else:
                earthquake_ReNaSSradius = 9`

 Distance on my website is in miles  ;-(   I dont know where I can change this  

Here is the template : 

https://meteo-pontscorff.fr/data-weewx/WMR200/belchertown/ 

Thanks  
Stef

**Comment by norecords** [comment]
> why you change this too ? line 1683 ?
> `if earthquake_maxradiuskm <= "1000": earthquake_ReNaSSradius = 5 else: earthquake_ReNaSSradius = 9`

That was a mistake and it's useless, just ignore it

**Comment by ddperso** [comment]
@steflp 
the 1st part is at about line 1690 (find "URL from Belchertown Weather"), it's for url
the 2nd part is at about line 1802 (find "self.generator.skin_dict["Extras"]["earthquake_server"] ") it's for data

**Comment by ddperso** [comment]
@norecords 
thank's for your tutorial, next time I'll try to create a pullrequest

it's a good idea to use earthquake_maxradius

Didier

**Comment by steflp** [comment]
Hello @norecords  and  @ddperso  
Ok for the few lines 
`if earthquake_maxradiuskm <= "1000": earthquake_ReNaSSradius = 5 else: earthquake_ReNaSSradius = 9`
I erase this  !

My last problem : I dont know how to change miles for Kms    ? 
Can you help me  ?  i dont find in any files ... 

Stephane

**Comment by norecords** [comment]
@steflp I think you missed a parameter in your weewx.conf or skin.conf like
```
       [[[Units]]]
            # This section is for managing the selection and formatting of units.

            [[[[Groups]]]]
....
                group_distance = km
.....
```

**Comment by steflp** [comment]
Very strange  I must change in belchertown.py   this : 
`distance_unit = converter.group_unit_dict["group_distance"]
 distance_unit = 'km'
`
 in tw file (skin.conf and weewx.conf) I  hve the good thing  ...
 It s ok  when i force it !
thanks

**Comment by norecords** [comment]
That mean your `group_distance` isn't set correctly..

**Comment by norecords** [comment]
@steflp this's now merged into master repo, you can update your weewx Belchertown's extension.
Parameters to be set in `weewx.conf` or in `skin.conf`
```
earthquake_enabled = 1
earthquake_server = ReNaSS
earthquake_maxradiuskm = 300 # as we are in France 300km is enought
```
You can adjust `belchertown.py` @ line 1698 the `minmagnitude` parameter to your needs.
For example, `minmagnitude=1.5`, but consider that under magnitude 2.0 this is imperceptible
ref: https://www.pyrenees-orientales.gouv.fr/content/download/5746/32255/file/seisme.pdf

**Comment by steflp** [comment]
Hello , 
Thanks a  lot...
for merging Belchertown file ,  i modify a lot of thing in this template , and , it will be difficult for me to update to the last at the moment.
I ve modified for minmagnitude to 2 in the url   ( i find it !)

`https://renass.unistra.fr/fdsnws/event/1/query?latitude=%s&longitude=%s&maxradius=%.2f&orderby=time&format=json&limit=1&minmagnitude=2&mindepth=1`

Thanks ^^

**Comment by steflp** [comment]
Hello
 Are you sure that this url  working good  ? 
`https://renass.unistra.fr/fdsnws/event/1/query?latitude=%s&longitude=%s&maxradius=%.2f&orderby=time&format=json&limit=1&minmagnitude=2&mindepth=1`

Its seems not OK 
Can you  confirm  ? 
Thanks 
Stéphane

**Comment by norecords** [comment]
hi @steflp 
yes, they change to the new api,
i'll have a look asap

**Comment by norecords** [comment]
@steflp see #602

**Comment by ddperso** [comment]
This issue is now closed by PR#602
